api/internal/localizer/locloader.go

```diff
@@ -89,9 +91,19 @@ func (ll *Loader) Load(path string) ([]byte, error) {
 	if err != nil {
 		return nil, errors.WrapPrefixf(err, "invalid file reference")
 	}
+
 	if filepath.IsAbs(path) {
-		return nil, errors.Errorf("absolute paths not yet supported in alpha: file path %q is absolute", path)
+		if !strings.HasPrefix(path, ll.Root()) {
+			return nil, fmt.Errorf("referenced absolute path '%s' in your Kustomization file escapes target directory", path)
```

Comment:
If I understand correctly, the user will be able to specify an absolute path but it should still be in the same tree as the root kustomization directory. If that is the case, could this message be closer to the one that already exists?

When I try to localize a path that is fully out-of-tree, I get the following error message (line breaks added by me for readability):
```
Error: unable to localize target ".":
  unable to localize configMapGenerator:
  unable to localize generator envs file:
  invalid file reference: security;
  file '/tmp/test.env' is not in or below '/foo/bar/kustomize-test'
```

It's not immediately clear to me what "escaping" means in this context, therefore I'd like to suggest a change:
```suggestion
			return nil, fmt.Errorf("referenced absolute path '%s' in your Kustomization file is not in the same tree as '%s'", path, ll.Root())
```

Also applies to the file below:
https://github.com/kubernetes-sigs/kustomize/blob/e3e4f873908b9ecf46eba4a9e312aae3dfe41d23/api/loader/fileloader.go#L184

Reply:
+1, thanks for this, I updated as suggested :)